src/OpenLoco/src/Windows/ObjectSelectionWindow.cpp

```diff
@@ -286,12 +310,31 @@ namespace OpenLoco::Ui::Windows::ObjectSelectionWindow
             != a.end();
     }
 
+    static VehicleType getVehicleTypeFromObject(TabObjectEntry& entry)
+    {
+        ObjectManager::freeTemporaryObject();
+        ObjectManager::loadTemporaryObject(*entry.object._header);
+
+        auto vehicleObj = reinterpret_cast<VehicleObject*>(ObjectManager::getTemporaryObject());
+        return vehicleObj->type;
+    }
```

Comment:
This is currently executed for every vehicle object in the list, and takes a good second to run. As this is run when the filter changes (whether clicking the tab or name pattern), this should probably just be done once. What would be a good place to cache the results?

Reply:
Could just add the vehicle type as a std::optional to the TabObjectEntry struct, I suppose. I'm thinking if not initialised, it's yet to be loaded, otherwise already available. Thoughts?